draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -3785,80 +3781,165 @@ Steps:
 7.  tv4 = CMOV(Z, -tv2, tv2 != 0)
 8.  tv4 = A * tv4
 9.  tv2 = tv3^2
-10. tv5 = tv4^2
-11. tv5 = A * tv5
+10. tv6 = tv4^2
+11. tv5 = A * tv6
 12. tv2 = tv2 + tv5
 13. tv2 = tv2 * tv3
-14. tv6 = tv4^3
+14. tv6 = tv6 * tv4
 15. tv5 = B * tv6
 16. tv2 = tv2 + tv5
 17.   x = tv1 * tv3
 18. (is_gx1_square, y1) = sqrt_ratio(tv2, tv6)
-19.   y = c1 * tv1
-20.   y = y * u
-21.   y = y * y1
-22.   x = CMOV(x, tv3, is_gx1_square)
-23.   y = CMOV(y, y1, is_gx1_square)
-24.  e1 = sgn0(u) == sgn0(y)
-25.   y = CMOV(-y, y, e1)
-26.   x = x / tv4
-27. return (x, y)
+19.   y = tv1 * u
+20.   y = y * y1
+21.   x = CMOV(x, tv3, is_gx1_square)
+22.   y = CMOV(y, y1, is_gx1_square)
+23.  e1 = sgn0(u) == sgn0(y)
+24.   y = CMOV(-y, y, e1)
+25.   x = x / tv4
+26. return (x, y)
 ~~~
 
-The sqrt_ratio subroutine used by the above procedure is defined immediately below.
-It depends on a constant S, which MUST be chosen to be a primitive element of the field F.
-See {{sswu-z-code}} for more information on identifying an appropriate value for S.
+### sqrt_ratio subroutines {#straightline-sswu-sqrt-ratio}
+
+This section defines three variants of the sqrt_ratio subroutine used by the
+above procedure.
+The first variant can be used with any field; the others are optimized versions
+for specific fields.
+
+The routines given in this section depend on the constant Z from the simplified SWU map.
```

Comment:
```suggestion
The routines given in this section depend on the constant Z from the simplified SWU map.
(The correctness of `map_to_curve_simple_swu` depends on using the same Z in `sqrt_ratio`.)
```

Reply:
Good catch. (I tried to hew to IETF-ish wording.)